drivers/network/tcpip/ip/transport/tcp/tcp.c

```diff
@@ -410,7 +410,16 @@ NTSTATUS TCPConnect
     Status = TCPTranslateError(LibTCPConnect(Connection,
                                                 &connaddr,
                                                 RemotePort));
-
+    if (Status != STATUS_SUCCESS && Status != STATUS_PENDING)
```

Comment:
Use `NT_SUCCESS` please instead of checking for STATUS_SUCCESS (this will conveniently also cover STATUS_PENDING).

Reply:
I listed them explicitly because I was unsure how some of the early failures would be treated by NT_SUCCESS.
NT_SUCCESS only returns true for > 0. Are you sure it would passes STATUS_PENDING but NOT return success for (ERR_RTE)STATUS_NETWORK_UNREACHABLE or (ERR_USE)STATUS_ADDRESS_ALREADY_EXISTS ??  

If you are sure I agree changing this would give better clarity.

Reply:
Actually.. there may also be later failures where this action might not be appropriate because the pcb is valid. I never reviewed the code to see if there were later failures where the pcb does actually need to be processed.

Reply:
added this change as well. I suppose there should be no harm in dropping those pending actions for all failures.

Reply:
The general NTSTATUS contract is that `NT_SUCCESS` codes mean success aka "output is valid", otherwise output is not valid. So unless this function is special (which it doesn't look like), this will (by design) behave exactly the way you'd want it to.